common/isolationgroup/interface.go

```diff
@@ -48,4 +51,7 @@ type State interface {
 
 	Subscribe(domainID, key string, notifyChannel chan<- ChangeEvent) error
 	Unsubscribe(domainID, key string) error
+
+	ProvideDomainCache(cache cache.DomainCache) State
```

Comment:
I don't think these methods are needed, because `domainCache` and `ConfigStore` may not be used by other implementation of this interface.

Reply:
This is a sequencing thing where I don't have access to DomainCache or ConfigStore when I'm calling the constructor.

I need to pass in the IsolationGroupState library in params, but at that point I don't have either the domain cache or the Config store. So this is the only way I could think to provide them in after they've been passed in as params.